### ISSUE TITLE
Two /workspace related issues

### ISSUE THREAD

**Issue by HansAdema** [heroku/pack-images#42]
Three months ago, in the middle of August, I was able to successfully build an image using Pack and the Heroku images. However, I just tried to build a new image today, but was unable to run the application due to various issues related to the `/workspace` directory.

I'm primarily using the official PHP buildpack, so my error story is based on that. You can verify this with the Heroku getting started example: https://github.com/heroku/php-getting-started

However, I also got errors trying to build the Python getting started image, so I assume all buildpacks are affected by this.

# Issue 1: $HOME is not set correctly

Immediately after building the example image, I am unable to start it.

```bash
$ pack build php-sample
[...]
$ docker run --rm php-sample
vendor/bin/heroku-php-apache2: line 260: php: command not found
This program requires PHP 5.5.11 or newer; check your 'php' command.
```

Checking the PATH inside the container, I see the following:

```
heroku@fb7a0fd3f366:/workspace$ echo $PATH
/home/heroku/.heroku/php/sbin:/home/heroku/.heroku/php/bin:/home/heroku/.heroku/php/sbin:/home/heroku/.heroku/php/bin:/home/heroku/.heroku/php/sbin:/home/heroku/.heroku/php/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/heroku/vendor/bin
```

Compare this to the PATH inside the previous, working application:

```
heroku@d384a0aaf899:~$ echo $PATH
/workspace/.heroku/php/sbin:/workspace/.heroku/php/bin:/workspace/.heroku/php/sbin:/workspace/.heroku/php/bin:/workspace/.heroku/php/sbin:/workspace/.heroku/php/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/workspace/vendor/bin
```

The large PATH is largely thanks to the PHP buildpack, which adds some search folders relative to `$HOME`. Which if you compare the old image:

```
heroku@d384a0aaf899:~$ echo $HOME
/workspace
```

Is indeed different from the new image:

```
heroku@fb7a0fd3f366:/workspace$ echo $HOME
/home/heroku
```

This appears to be a consequence of this change: https://github.com/heroku/pack-images/commit/28e9a2cda3d8025db0614a31326599821e0f467d#diff-6f2bd57d9bf5b098ff9f4b368b035e49L12

# Issue 2: The `/app` symlink is missing

To fix the issue above, I built a custom version of the `pack` image which restores the homedir argument from the breaking change. This however lead me to a new version:

```
httpd: error while loading shared libraries: libaprutil-1.so.0: cannot open shared object file: No such file or directory
This program requires Apache 2.4.10 or newer with mod_proxy and mod_proxy_fcgi enabled; check your 'httpd' command.
```

Checking if this really is the case:

```
heroku@2af983eaebaf:~$ which httpd
/workspace/.heroku/php/sbin/httpd
heroku@2af983eaebaf:~$ ldd /workspace/.heroku/php/sbin/httpd
	linux-vdso.so.1 (0x00007ffc6f35e000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007ff04ae08000)
	libaprutil-1.so.0 => not found
	libapr-1.so.0 => not found
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ff04abe9000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ff04a7f8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007ff04b324000)
```

As you can see, some of the libraries are missing.

Comparing this to the previous image:

```
heroku@d384a0aaf899:~$ which httpd
/workspace/.heroku/php/sbin/httpd
heroku@d384a0aaf899:~$ ldd /workspace/.heroku/php/sbin/httpd
	linux-vdso.so.1 (0x00007ffcf9384000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f0fd4893000)
	libaprutil-1.so.0 => /app/.heroku/php/lib/libaprutil-1.so.0 (0x00007f0fd4669000)
	libapr-1.so.0 => /app/.heroku/php/lib/libapr-1.so.0 (0x00007f0fd4432000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f0fd4213000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f0fd3e22000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f0fd3bf0000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f0fd39b8000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f0fd37b4000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f0fd4daf000)
```

As you can see, the HTTPD server as built by the PHP buildpack looks in the `/app` folder. But Cloud Native Buildpacks don't use the `/app` folder but the `/workspace` folder. And you can actually see that the libraries are correctly placed in the `/workspace/.heroku/php/lib/` folder.

In the past, this worked because a symlink was created between `/workspace` and `/app`. But it appears that this change removed this symlink.

https://github.com/heroku/pack-images/commit/28e9a2cda3d8025db0614a31326599821e0f467d#diff-6f2bd57d9bf5b098ff9f4b368b035e49L3

# Version Report

Pack: `v0.5.0 (git sha: c9cfac75b49609524e1ea33f809c12071406547c)`
Docker: `Docker version 19.03.2, build 6a30dfca03`
Run image: `heroku/pack:18@b5c992735c7b`
Build image `heroku/buildpacks:18@eb25208e1361`

**Comment by jkutner** [comment]
@HansAdema I've merged #43 and published `heroku/buildpacks:18`. Can you confirm that this works now?

Thanks for your patience through the freeze.

**Comment by edmorley** [comment]
Hi!

I believe this issue has been resolved for a while - since the PHP getting started guide builds/runs fine for me locally (it's also tested in CI now, albeit just that it builds/boots:
https://app.circleci.com/pipelines/github/heroku/pack-images/1703/workflows/bb24bf93-68fb-4e8b-9851-9c76ac7d4acb/jobs/34014).

I believe this was fixed via a combination of #43, #54 and potentially some of the improvements to [cnb-shim](https://github.com/heroku/cnb-shim).
